x-pack/plugins/apm/server/lib/rum_client/get_long_task_metrics.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  getRumLongTasksProjection,
+  getRumOverviewProjection,
+} from '../../projections/rum_overview';
+import { mergeProjection } from '../../projections/util/merge_projection';
+import {
+  Setup,
+  SetupTimeRange,
+  SetupUIFilters,
+} from '../helpers/setup_request';
+import { SPAN_DURATION } from '../../../common/elasticsearch_fieldnames';
+
+export async function getLongTaskMetrics({
+  setup,
+}: {
+  setup: Setup & SetupTimeRange & SetupUIFilters;
+}) {
+  const projection = getRumLongTasksProjection({
+    setup,
+  });
+
+  const params = mergeProjection(projection, {
+    body: {
+      size: 0,
+      query: {
+        bool: projection.body.query.bool,
+      },
+      aggs: {
+        transIds: {
+          terms: {
+            field: 'transaction.id',
+            size: 1000,
+          },
+          aggs: {
+            sumLongTask: {
+              sum: {
+                field: SPAN_DURATION,
+              },
+            },
+            longestLongTask: {
+              max: {
+                field: SPAN_DURATION,
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+
+  const { apmEventClient } = setup;
+
+  const response = await apmEventClient.search(params);
+  const { transIds } = response.aggregations ?? {};
+
+  const validTransactions: string[] = await filterPageLoadTransactions(
+    setup,
+    (transIds?.buckets ?? []).map((bucket) => bucket.key as string)
+  );
+  let noOfLongTasks = 0;
+  let sumOfLongTasks = 0;
+  let longestLongTask = 0;
+
+  (transIds?.buckets ?? []).forEach((bucket) => {
+    if (validTransactions.includes(bucket.key as string)) {
+      noOfLongTasks += bucket.doc_count;
+      sumOfLongTasks += bucket.sumLongTask.value ?? 0;
+      if ((bucket.longestLongTask.value ?? 0) > longestLongTask) {
+        longestLongTask = bucket.longestLongTask.value!;
+      }
+    }
+  });
+  return {
+    noOfLongTasks,
+    sumOfLongTasks,
+    longestLongTask,
+  };
+}
+
+async function filterPageLoadTransactions(
```

Comment:
One thing I'd say for both of these ES callers is we should add some unit tests, if possible. There are some minor conditionals and it's all pretty straightforward, but I am wary of merging too much untested code. I know we have E2E tests here as well, and if you think they provide sufficient coverage that is probably ok for now. Nevertheless, we should be cognizant of the code volume we create without associated unit tests, too, as they can be more granular and are a lot cheaper to maintain.